### PR TITLE
Admonition directives

### PR DESCRIPTION
**Pull request by jonas**
screen shot 2016-10-27 at 10 36 57 pm

### CODE REVIEW

**Comment by jonas** [comment]
A WIP branch to use this in Akka HTTP: jonas/akka-http@9e8c0bec0deb72dd95f6b2896af3d060f5b7a396

**Comment by ktoso** [comment]
Awesome! :-)

File: core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala [hunk 1]
```diff
@@ -366,3 +366,20 @@ case class VarsDirective(variables: Map[String, String]) extends ContainerBlockD
     }
   }
 }
+
+/**
+ * Adminition directive.
```
Comment (dwijnand) [inline core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala]: Admonition is misspelt

Comment (jonas) [inline core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala]: Thanks, fixed

File: docs/src/main/paradox/features/additional-directives.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+Additional directives
```
Comment (eed3si9n) [inline docs/src/main/paradox/features/additional-directives.md]: We need a better name here, otherwise this page either will be kitchen sink of directives, or we'll have `more-additional-directives.md`. Suggestion is "Callout box" `callout-box.md`. Who knows what a callout is? Hey it's better than "Additional directives".

File: docs/src/main/paradox/features/linking.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Linking
 -------
 
-### @ref link
+## @ref link
```
Comment (eed3si9n) [inline docs/src/main/paradox/features/linking.md]: Isn't Linking with `---` an h2? Why is `@ref` getting a promotion to h2 here?

File: docs/src/main/paradox/features/templating.md [hunk 4]
```diff
@@ -3,29 +3,48 @@
 Templating
 ----------
 
-Paradox uses [StringTemplate][st] for the basic templating. For example:
+Paradox uses [StringTemplate][st] for the basic templating. Template files have
```
Comment (eed3si9n) [inline docs/src/main/paradox/features/templating.md]: I think this commit should be split into another PR.

File: core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala [hunk 5]
```diff
@@ -366,3 +366,20 @@ case class VarsDirective(variables: Map[String, String]) extends ContainerBlockD
     }
   }
 }
+
+/**
+ * Admonition directive.
+ *
+ * Replaces property values in verbatim blocks.
+ */
+case class AdmonitionDirective(name: String, defaultTitle: String) extends ContainerBlockDirective(Array(name): _*) {
```
Comment (eed3si9n) [inline core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala]: What do you think about renaming to CalloutDirective?

Comment (jonas) [inline core/src/main/scala/com/lightbend/paradox/markdown/Directive.scala]: At least blame Sphinx!

File: docs/src/main/paradox/features/templating.md (hunk 4 above)
Comment (jonas) [inline docs/src/main/paradox/features/templating.md]: Agreed.

File: docs/src/main/paradox/features/linking.md (hunk 3 above)
Comment (jonas) [inline docs/src/main/paradox/features/linking.md]: Yes, true. My main goal was to make the side menu more useful.

**Comment by dwijnand** [comment]
I need to find a way to hold your PR hostage so you can ship me https://github.com/jonas/tig/issues/313 :-P

File: docs/src/main/paradox/features/additional-directives.md (hunk 2 above)
Comment (jonas) [inline docs/src/main/paradox/features/additional-directives.md]: Sorry I didn't do a better job at researching a good name. Turns out that foundation has callouts.

**Comment by jonas** [comment]
Let me know if I should leave foundation's background color for warnings.

screen shot 2016-10-26 at 10 26 27 pm

**Comment by jonas** [comment]
@dwijnand I got the rendering part working parsing standout control characters emitted by `diff-highlight`. I'll see if I can find some time to revive that patch. :-)

**Comment by eed3si9n** [comment]
As per color palette, my policy around the "generic" theme was to make it feel neutral as possible, which means make it kind of look like Github's documentation. See their warning and note callouts here:
- https://developer.github.com/guides/using-ssh-agent-forwarding/
- https://developer.github.com/v3/gists/#create-a-gist

**Comment by sirthias** [comment]
Awesome stuff! Really looking forward to this!

**Comment by pvlugter** [comment]
Nice one!

**Comment by jonas** [comment]
@eed3si9n I personally find the warning colors a bit too dark which is why [Foundation's](http://foundation.zurb.com/sites/docs/callout.html) warning color was left as is.

Some examples: http://codepen.io/anon/pen/RGmgRx
I can add the pure version of GitHub and Foundation's callout if you want.
screen shot 2016-10-27 at 10 20 12 pm
Note, I probably won't have time to play with colors before Monday, so let me know if I should split out the 3 first commits for a release of 0.2.6 without callouts.

**Comment by eed3si9n** [comment]
#3 looks good. Thanks for working on this.
